presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -93,6 +97,59 @@ public JoinNode(@JsonProperty("id") PlanNodeId id,
         checkArgument(!isCrossJoin() || inputSymbols.equals(outputSymbols), "Cross join does not support output symbols pruning or reordering");
     }
 
+    public JoinNode flip()
+    {
+        return new JoinNode(
+                getId(),
```

Comment:
I imagine any time you're flipping a join, you'll use only one or the other.  But a no duplicate ids verifier is a good idea in any case.